CHANGELOG.md

```diff
@@ -28,6 +28,14 @@
         * `Iterable<Byte>.decodeAsn1VarUInt()`
         * `ByteArray.decodeAsn1VarUInt()`
 * Revamp implicit tagging
+* Consume only the first `Asn1Element.parse()` only consumes the first parsable element and
```

Comment:
i would consider this change _very_ carefully: if I understand correctly, before this change, `parse()` guaranteed that there are no trailing bytes. now, you would need to call `parseWithRemainder` and then manually inspect the returned remainder to ensure it is empty.

if my understanding is correct, this could have major impact on strict validation, where trailing bytes are now quietly ignored. do we really want this? or do we want `parse` to be `parseWithRemainder().let { (elm, remainder) -> assert(remainder.isEmpty()); elm }`, so user code needs to explicitly opt into allowing a tail by using `parseWithRemainder()`?

Reply:
good point! agree

Reply:
reworked